### PR TITLE
Add v6 odhcpd and dhcp6c which are now part of openwrt base upstream

### PR DESCRIPTION
**Pull request by cshoredaniel**
odhcpd is the default DHCPv6(-PD)/RA server and relay in openwrt-core, so add it to openwrt-minimal.  Also make sure it works by having dnsmasq be sure to ship /etc/config/dhcp (which odhcpd also relies on; seems a bit excessive to make this single file a common package though).

### CODE REVIEW

File: recipes-core/odhcpd/odhcpd_git.bb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+SRC_URI = "\
+    git://git.openwrt.org/project/odhcpd.git;name=odhcpd \
+"
+
+S = "${WORKDIR}/git"
+OF = "${S}/openwrt/package/network/services/odhcpd/files"
+
+EXTRA_OECMAKE_append = " -DUBUS=1 -Dlibnl=${STAGING_LIBDIR}/libnl-3.so -Dlibnl-tiny_include_dir=${STAGING_INCDIR}/libnl3"
+
+OECMAKE_C_FLAGS += "-Wno-error=cpp"
```
Comment (kraj) [inline recipes-core/odhcpd/odhcpd_git.bb]: why is this needed ?

File: recipes-core/odhcpd/odhcpd_git.bb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+SRCREV_openwrt = "${OPENWRT_SRCREV}"
+
+SRC_URI = "\
+    git://git.openwrt.org/project/odhcpd.git;name=odhcpd \
+"
+
+S = "${WORKDIR}/git"
+OF = "${S}/openwrt/package/network/services/odhcpd/files"
+
+EXTRA_OECMAKE_append = " -DUBUS=1 -Dlibnl=${STAGING_LIBDIR}/libnl-3.so -Dlibnl-tiny_include_dir=${STAGING_INCDIR}/libnl3"
```
Comment (kraj) [inline recipes-core/odhcpd/odhcpd_git.bb]: hmm hardcoded path to a library seems a bit odd. Yocto used sysroot to find target libs is that not working ?

File: recipes-core/odhcpd/odhcpd_git.bb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+OECMAKE_C_FLAGS += "-Wno-error=cpp"
+
+FILES_${PN} += "\
+    ${sysconfdir}/uci-defaults/odhcpd.defaults \
+    ${sysconfdir}/init.d/odhcpd \
+    ${sbindir}/odhcpd-update \
+"
```
Comment (kraj) [inline recipes-core/odhcpd/odhcpd_git.bb]: move FILES_${PN} below do_install see https://www.openembedded.org/wiki/Styleguide

File: recipes-core/odhcpd/odhcpd_git.bb (hunk 2 above)
Comment (cshoredaniel) [reply recipes-core/odhcpd/odhcpd_git.bb]: no it's not; problem is the recipe is called libnl but the library is libnl-3 - the PROVIDES goes off the recipe name not the library name so Yocto doesn't find nl-3 - I guess that could be considered a bug in core.

File: recipes-core/odhcpd/odhcpd_git.bb (hunk 1 above)
Comment (cshoredaniel) [reply recipes-core/odhcpd/odhcpd_git.bb]: hmmm....it might not be, that was left over from copy&paste.

File: recipes-core/odhcpd/odhcpd_git.bb (hunk 3 above)
Comment (cshoredaniel) [reply recipes-core/odhcpd/odhcpd_git.bb]: will do

File: recipes-core/odhcpd/odhcpd_git.bb (hunk 2 above)
Comment (kraj) [reply recipes-core/odhcpd/odhcpd_git.bb]: will something like
```
-Dlibnl=-lnl-3
```

work ?

Comment (cshoredaniel) [reply recipes-core/odhcpd/odhcpd_git.bb]: I'll try that -- I tried other things for -Dlibnl but I guess it was the wrong syntax.

Comment (cshoredaniel) [reply recipes-core/odhcpd/odhcpd_git.bb]: didn't work, get build/tmp-glibc/work/armv5e-oe-linux-gnueabi/odhcpd/git-r0/build'
| make[2]: *** No rule to make target '-lnl-3', needed by 'odhcpd'.  Stop.

Looking at odhcpd's CMakeLists.txt it's kind of funky I think.  Probably need to patch it.

The issue is that upstream has libnl-tiny and that's assumption goes into CMakeLists.txt

**Comment by cshoredaniel** [comment]
Ok they should be *much* better now!
